### PR TITLE
DDI-1352: Left side menu - 1st version - Angular

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
### Description:
Side menu (collaboration with CS platform team) - First version - Angular supported.

### Changes:

- Sidebar and SidebarItem svelte components added
- Handle multi-level items. However, for now, the design (CSS styles) is supported for 2-level menu items. 
- Unit tests added 
- Angular demo app added 

### Screenshots:
This is our Angular demo app:

image

This is the demo site:
https://vanessatran-ddi.github.io/angular-ui-component-sample/home
image

This is the accessibility feature that works: 

https://github.com/GovAlta/ui-components/assets/120135417/897b7af6-53d7-45ee-86f0-0a6a7f3a15ae

### Todo next (Another PR will follow up):

- [ ] React wrapper 
- [ ] Storybook

### CODE REVIEW

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  li.active {
+    background: var(--goa-color-info-background);
+  }
+
+  li:focus {
+    outline: var(--goa-border-width-l) solid var(--goa-color-interactive-focus);
+  }
+
+  /*Sub Item Style*/
+  li.subLevel {
```
Comment (chrisolsen) [inline libs/web-components/src/components/left-side-menu/LeftNav.svelte:209]: Can we stick to the class name convention of `sub-level` instead?

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    padding: 10px 12px 1vh 16px;
```
Comment (chrisolsen) [inline libs/web-components/src/components/left-side-menu/LeftNav.svelte:210]: Are these padding value correct? Having four different values is a little odd.

What is the reason `1vh` is being used? (1% of the vertical height)

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    margin: 0;
+  }
+
+  li {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    padding: 10px var(--goa-space-s) 1vh 32px;
+    gap: 18px;
```
Comment (chrisolsen) [inline libs/web-components/src/components/left-side-menu/LeftNav.svelte:177]: Should be using `rem` values. This one also has some inconsistent padding. 

file:///home/chris/Pictures/Screenshot_20230524_100956.png

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    overflow: hidden;
```
Comment (chrisolsen) [inline libs/web-components/src/components/left-side-menu/LeftNav.svelte:178]: if overflow is hidden, the text should styled to show the `...`
```
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
```

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    displayname: string;
+    name?: string;
+    active?: boolean;
+    url?: string;
+    items?: NavItem[];
+  }
+
+  // Public props, needed for recursion
+  export let level = 0;
+  export let navItems = [];
```
Comment (chrisolsen) [inline libs/web-components/src/components/left-side-menu/LeftNav.svelte:16]: For exported properties the name needs to be all downcased when compiling to web-components

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 6]
```diff
@@ -0,0 +1,225 @@
+<svelte:options tag="goa-left-nav"/>
```
[the remark this answers is not in the document]
Comment (ArakTaiRoth) [reply libs/web-components/src/components/left-side-menu/LeftNav.svelte:1]: @chrisolsen Yes, that's the name agreed upon between us and the Child Services team

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    if (_rootElement) {
+      _items = level === 0 ? getItems() : navItems;
+      console.log(_items);
+    }
+  }
+
+  onMount(async () => {
+    const slot = _rootElement.querySelector("slot");
+    slot?.addEventListener("slotchange", (_e) => {
+      _items = level === 0 ? [...getItems()] : [...navItems];
```
[the remark this answers is not in the document]
Comment (vanessatran-ddi) [reply libs/web-components/src/components/left-side-menu/LeftNav.svelte:33]: @chrisolsen Yeah it doesn't work, especially for level > 0 (expand menu). I need to do [...array] to create a shallow object.

File: libs/web-components/src/components/left-side-menu/LeftNav.svelte (hunk 2 above)
Comment (vanessatran-ddi) [reply libs/web-components/src/components/left-side-menu/LeftNav.svelte:210]: Yes, I need to use 1vh to make the text vertically center (our text has a line-height)
For the padding they are correct according to the designer's verification.

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**